tests/test_tokamak_source.py

```diff
@@ -1,74 +1,117 @@
+    assume(minor_radius > pedestal_radius)
+    assume(minor_radius > shafranov_factor)
 
+    args_dict = {
+        "elongation" : elongation,
+        "triangularity" : triangularity,
+        "major_radius" : major_radius,
+        "minor_radius" : minor_radius,
+        "pedestal_radius" : pedestal_radius,
+        "shafranov_factor" : shafranov_factor,
+        "ion_density_centre" : 1.09e20,
+        "ion_density_peaking_factor" : 1,
+        "ion_density_pedestal" : 1.09e20,
+        "ion_density_separatrix" : 3e19,
+        "ion_temperature_centre" : 45.9,
+        "ion_temperature_peaking_factor" : 8.06,
+        "ion_temperature_pedestal" : 6.09,
+        "ion_temperature_separatrix" : 0.1,
+        "mode" : "H",
+        "ion_temperature_beta" : 6,
+    }
 
-def test_creation():
-    my_source = TokamakSource(
-        elongation=1.557,
-        ion_density_centre=1.09e20,
-        ion_density_peaking_factor=1,
-        ion_density_pedestal=1.09e20,
-        ion_density_separatrix=3e19,
-        ion_temperature_centre=45.9,
-        ion_temperature_peaking_factor=8.06,
-        ion_temperature_pedestal=6.09,
-        ion_temperature_separatrix=0.1,
-        major_radius=9.06,
-        minor_radius=2.92258,
-        pedestal_radius=0.8 * 2.92258,
-        mode="H",
-        shafranov_factor=0.44789,
-        triangularity=0.270,
-        ion_temperature_beta=6,
-    )
-    for source in my_source.make_openmc_sources():
+    return args_dict if return_dict else TokamakSource(**args_dict)
+
+
+@given(tokamak_source=tokamak_source_strategy())
+@settings(max_examples=1, deadline=None)
+def test_creation(tokamak_source):
+    """Tests that the sources generated by TokamakSource are of type openmc.Source"""
+    for source in tokamak_source.sources:
         assert isinstance(source, Source)
 
 
-def test_strengths_are_normalised():
+@given(tokamak_source=tokamak_source_strategy())
+@settings(max_examples=100, deadline=None)
+def test_strengths_are_normalised(tokamak_source):
     """Tests that the sum of the strengths attribute is equal to"""
-    my_source = TokamakSource(
-        elongation=1.557,
-        ion_density_centre=1.09e20,
-        ion_density_peaking_factor=1,
-        ion_density_pedestal=1.09e20,
-        ion_density_separatrix=3e19,
-        ion_temperature_centre=45.9,
-        ion_temperature_peaking_factor=8.06,
-        ion_temperature_pedestal=6.09,
-        ion_temperature_separatrix=0.1,
-        major_radius=9.06,
-        minor_radius=2.92258,
-        pedestal_radius=0.8 * 2.92258,
-        mode="H",
-        shafranov_factor=0.44789,
-        triangularity=0.270,
-        ion_temperature_beta=6,
-    )
-    assert pytest.approx(sum(my_source.strengths), 1)
-
-
-def test_angles():
+    assert pytest.approx(sum(tokamak_source.strengths), 1)
+
+
+@given(tokamak_source=tokamak_source_strategy())
+@settings(max_examples=100, deadline=None)
+def test_source_locations_are_within_correct_range(tokamak_source):
```

Comment:
Nice one! 

I don't see the shafranov shift in this test, shouldn't it appear somewhere?

Reply:
I didn't think the Shafranov shift affected the shape of the last closed magnetic surface, as in the equations for RZ that term equals 0 when a=A, and this test only checks that R is within that outer bound.

Reply:
You are right!